src/main/kotlin/no/nav/familie/tilbake/behandlingskontroll/BehandlingskontrollService.kt

```diff
@@ -53,31 +56,56 @@ class BehandlingskontrollService(private val behandlingsstegstilstandRepository:
         behandletSteg.map { behandlingsstegstilstandRepository.update(it.copy(behandlingsstegsstatus = AVBRUTT)) }
-        oppdaterBehandlingsstegsstaus(behandlingId, behandlingsstegMedStatus)
+        oppdaterBehandlingsstegsstaus(behandlingId, behandlingsstegsinfo)
     }
 
     @Transactional
-    fun oppdaterBehandlingsstegsstaus(behandlingId: UUID, behandlingsstegMedStatus: BehandlingsstegMedStatus) {
+    fun oppdaterBehandlingsstegsstaus(behandlingId: UUID, behandlingsstegsinfo: Behandlingsstegsinfo) {
         val behandling = behandlingRepository.findByIdOrThrow(behandlingId)
         if (behandling.erAvsluttet()) {
             throw Feil("Behandling med id=$behandlingId er allerede ferdig behandlet, " +
-                       "så status=${behandlingsstegMedStatus.behandlingsstegstatus} kan ikke oppdateres")
+                       "så status=${behandlingsstegsinfo.behandlingsstegstatus} kan ikke oppdateres")
         }
         val behandlingsstegstilstand = behandlingsstegstilstandRepository.findByBehandlingIdAndBehandlingssteg(
-                behandlingId, behandlingsstegMedStatus.behandlingssteg)
+                behandlingId, behandlingsstegsinfo.behandlingssteg)
                                        ?: throw Feil(message = "Behandling med id=$behandlingId og " +
-                                                               "steg=${behandlingsstegMedStatus.behandlingssteg} finnes ikke")
+                                                               "steg=${behandlingsstegsinfo.behandlingssteg} finnes ikke")
         behandlingsstegstilstandRepository.update(behandlingsstegstilstand.copy(
-                behandlingsstegsstatus = behandlingsstegMedStatus.behandlingsstegstatus))
+                behandlingsstegsstatus = behandlingsstegsinfo.behandlingsstegstatus,
+                venteårsak = behandlingsstegsinfo.venteårsak,
+                tidsfrist = behandlingsstegsinfo.tidsfrist,
+        ))
+    }
+
+    @Transactional
+    fun settBehandlingPåVent(behandlingId: UUID, venteårsak: Venteårsak, tidsfrist: LocalDate) {
+        val behandlingsstegstilstand: List<Behandlingsstegstilstand> =
+                behandlingsstegstilstandRepository.findByBehandlingId(behandlingId)
+        val aktivtBehandlingsstegstilstand = finnAktivStegstilstand(behandlingsstegstilstand)
+                                             ?: throw Feil(message = "Behandling $behandlingId " +
+                                                                     "har ikke aktivt steg",
+                                                           frontendFeilmelding = "Behandling $behandlingId " +
+                                                                                 "har ikke aktivt steg")
+        if (VENTER == aktivtBehandlingsstegstilstand.behandlingsstegsstatus &&
+            aktivtBehandlingsstegstilstand.tidsfrist?.isAfter(tidsfrist) == true) {
+            throw Feil(message = "Fristen kan ikke senkes for behandling $behandlingId",
```

Comment:
Er det noen grunn for at det ikke skal være mulig å korte ned fristen? Hva om saksbehandler har registrert feil manuellt?

Reply:
Det fikset jeg. Den kan ikke være mindre enn dagens dato.